install.sh

```diff
@@ -1,354 +1,287 @@
-# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-# SOFTWARE.
-#
-#
-#  - https://github.com/kvz/bash3boilerplate
-#  - http://kvz.io/blog/2013/02/26/introducing-bash3boilerplate/
-#
-# Version: 2.0.0
-#
-# Authors:
-#
-#  - Kevin van Zonneveld (http://kvz.io)
-#  - Izaak Beekman (https://izaakbeekman.com/)
-#  - Alexander Rathai (Alexander.Rathai@gmail.com)
-#  - Dr. Damian Rouson (http://www.sourceryinstitute.org/) (documentation)
-#
-# Licensed under MIT
-# Copyright (c) 2013 Kevin van Zonneveld (http://kvz.io)
-
-# The invocation of bootstrap.sh below performs the following tasks:
-# (1) Import several bash3boilerplate helper functions & default settings.
-# (2) Set several variables describing the current file and its usage page.
-# (3) Parse the usage information (default usage file name: current file's name with -usage appended).
-# (4) Parse the command line using the usage information.
-
-### Start of boilerplate -- do not edit this block #######################
-export OPENCOARRAYS_SRC_DIR="${OPENCOARRAYS_SRC_DIR:-${PWD%/}}"
-if [[ ! -f "${OPENCOARRAYS_SRC_DIR}/src/libcaf.h" ]]; then
-  echo "Please run this script inside the top-level OpenCoarrays source directory or "
-  echo "set OPENCOARRAYS_SRC_DIR to the OpenCoarrays source directory path."
-  exit 1
-fi
-export B3B_USE_CASE="${B3B_USE_CASE:-${OPENCOARRAYS_SRC_DIR}/prerequisites/use-case}"
-if [[ ! -f "${B3B_USE_CASE:-}/bootstrap.sh" ]]; then
-  echo "Please set B3B_USE_CASE to the bash3boilerplate use-case directory path."
-  exit 2
-else
-    # shellcheck source=./prerequisites/use-case/bootstrap.sh
-    source "${B3B_USE_CASE}/bootstrap.sh" "$@"
-fi
-### End of boilerplate -- start user edits below #########################
+#!/bin/sh
 
-# Set expected value of present flags that take no arguments
-export __flag_present=1
+set -e # exit on error
 
-# Set up a function to call when receiving an EXIT signal to do some cleanup. Remove if
-# not needed. Other signals can be trapped too, like SIGINT and SIGTERM.
-function cleanup_before_exit () {
-  info "Cleaning up. Done"
+print_usage_info()
+{
+    echo "OpenCoarrays Installation Script"
+    echo ""
+    echo "USAGE:"
+    echo "./install.sh [--help | [--prefix=PREFIX]"
+    echo ""
+    echo " --help             Display this help text"
+    echo " --prefix=PREFIX    Install binary in 'PREFIX/bin'"
+    echo " --prereqs          Display a list of prerequisite software."
+    echo "                    Default prefix='\$HOME/.local/bin'"
+    echo ""
+    echo "For a non-interactive build with the 'yes' utility installed, execute"
+    echo "yes | ./install.sh"
```

Comment:
@everythingfunctional I was going to remove this, but decided to keep it because it relates to a [user request].  I think I can borrow [code from the original installer] to cover the case in which `sudo` is required.

[user request]: https://github.com/sourceryinstitute/OpenCoarrays/issues/191
[code from the original installer]: https://github.com/sourceryinstitute/OpenCoarrays/blob/main/prerequisites/build-functions/set_SUDO_if_needed_to_write_to_directory.sh